.github/workflows/build.yml

```diff
@@ -167,6 +167,173 @@ jobs:
+          echo "🔄 Run attempt: ${{ github.run_attempt }}"
+          
+          # Force complete cleanup of any cached data
+          rm -rf ~/.cache/trivy || true
+          rm -rf /tmp/trivy* || true
+          
+          # Download fresh vulnerability database
+          trivy image --download-db-only
+          
+          echo "✅ Trivy database updated successfully"
+          echo "📊 Database info:"
+          trivy version --format json | jq '.VulnerabilityDB' || trivy version
+          
+          # Show database timestamp to detect any caching issues
+          echo "📅 Database cache info:"
+          ls -la ~/.cache/trivy/ 2>/dev/null || echo "No cache directory found"
+
+      - name: Verify Trivy installation
+        run: |
+          trivy version
+          echo "Trivy installed successfully"
+
+      - name: Show scan configuration
+        run: |
+          echo "🔧 Security Scan Configuration:"
+          echo "  - Scanning: Production dependencies"
+          echo "  - Severity levels: CRITICAL, HIGH, MEDIUM, LOW"
+          echo "  - Skipping directories: common/temp"
+          echo "  - Timeout: 10 minutes"
+          echo "  - Exit on vulnerabilities: YES (Build will fail)"
+
+      - name: Run Trivy vulnerability scanner
+        run: |
+          echo "🔍 Scanning for security vulnerabilities..."
+          echo "🔄 Run attempt: ${{ github.run_attempt }}"
+          echo "📂 Current working directory: $(pwd)"
+          echo ""
+          
+          # Debug: Check environment and file system state
+          echo "📊 Environment check:"
+          echo "  - Build artifacts restored: $(test -f rush.json && echo 'YES' || echo 'NO')"
+          echo "  - Rush installed: $(rush --version 2>/dev/null && echo 'YES' || echo 'NO')"
+          echo "  - ui-toolkit exists: $(test -d workspaces/common-libs/ui-toolkit && echo 'YES' || echo 'NO')"
+          echo "  - ui-toolkit node_modules: $(test -d workspaces/common-libs/ui-toolkit/node_modules && echo 'YES' || echo 'NO')"
+          echo ""
+          
+          # Check if axios is actually installed with vulnerable version
+          if [ -d "workspaces/common-libs/ui-toolkit/node_modules/axios" ]; then
+            AXIOS_VERSION=$(cat workspaces/common-libs/ui-toolkit/node_modules/axios/package.json | grep '"version"' | cut -d'"' -f4)
+            echo "  - axios version found: $AXIOS_VERSION"
+            if [ "$AXIOS_VERSION" = "0.21.0" ]; then
+              echo "  - ⚠️  VULNERABLE AXIOS VERSION DETECTED: $AXIOS_VERSION"
+            fi
+          else
+            echo "  - axios not found in node_modules"
+          fi
+          echo ""
+          
+          # Run Trivy vulnerability scan
+          echo "🚀 Starting Trivy scan..."
+          trivy fs . \
+            --timeout 10m \
+            --skip-dirs common/temp \
+            --severity CRITICAL,HIGH,MEDIUM,LOW \
+            --format table \
+            --output trivy-results.txt \
+            --exit-code 0
```

Comment:
Logic can be simplified by removing the forced exit code and using a separate step for failure reporting.